Edits to PGD PR5172 (#5263)
EnterpriseDB/docs: product_docs/docs/pgd/5/node_management/maintainance_with_proxies.mdx

## Maintenance and performance

> As a general rule, don't perform maintenance operations on a cluster's write leader. 

Reviewer: No - You Should Never. This is very important emphasis.

Author: ```suggestion
As a general rule, you should never perform maintenance operations on a cluster's write leader. 
```

Maintenance operations such as `VACUUM` can be quite disruptive to the smooth running of a busy server and often detrimental to workload performance.
Therefore, it's best to run maintenance commands on any node in a group that isn't the write leader.

> Generally, this requires you to connect directly and issue the maintenance commands on the non-write-leader nodes.

Reviewer: This feels imprecise with compared to There are some situations.

Author: "There are" is a weak sentence starter (or in this case phrase starter). I don't think it's any less precise. You can revert it if you feel strongly about it.

But in some situations, this isn't possible.

## Maintenance and proxies

Proxies, by design, always connect to and send commands to the current write leader.

> This usually means that you must not connect to a proxy to perform maintenance.

Reviewer: Changed the meaning. You connect through or via a proxy. You do not connect to a proxy.

Author: ```suggestion
This usually means that you must not connect by way of a proxy to perform maintenance.
```

Sorry, that was an error on my part. I meant to change "via" to "by way of."

PGD clusters nodes can present a direct connection for psql and PGD CLI clients that you can use to issue maintenance commands to the server on those nodes. 
But there are environment in which the PGD cluster is deployed where a proxy is the only way to access the cluster.

For example, in BigAnimal, PGD clusters are locked down such that the only access to the database is through an instance of PGD Proxy. 
This configuration reduces the footprint of the cluster and makes it more secure. However, it requires that you use a different way of sending maintenance requests to the cluster's nodes.

The technique outlined here is generally useful for despatching commands to specific nodes without being directly connected to that node's server.

## Maintenance commands

The term *maintenance commands* refers to:

* `VACUUM` 

> * Non-replicated DDL commands (which you might want to manually replicate)

Reviewer: May is for something factual which is possible
Might is for something counter factual or unlikely to be possible

Author: "May" is ambiguous and according to tech writing standards it should be replaced with "might" or "can," whichever is more accurate. I don't think the words have the meanings you are suggesting. In this case, may doesn't cause any ambiguity and I've replaced it out of habit. Feel free to revert if you feel the meaning will be clearer.

Reviewer: https://www.grammarly.com/blog/may-might/ (and various other places)

Reviewer 2: If we use can, and then clarify this is just optional? For example: 
```suggestion
* Non-replicated DDL commands (which you can replicate manually if required)
```

## A note on node names

> The servers in the cluster are referred to by their PGD cluster node names. To get a list of node names in your cluster, use:

Reviewer: You seem to have made this passive

Author: We don't write in first person, and I was having a hard time revising to eliminate first person without going passive. Feel free to suggest an edit that doesn't use first person.

Reviewer: In this article, the servers in the cluster are identified by their PGD cluster node names.

Author: ```suggestion
In this discussion, the servers in the cluster are identified by their PGD cluster node names. To get a list of node names in your cluster, use:
```
What do you think of "discussion" in place of article? It's not really an article, it's part of the documentation.

Reviewer 2: What do you think about this? 

```suggestion
This section uses PGD cluster node names to refer to servers in the cluster. To get a list of node names in your cluster, use:
```

```SQL
select node_name from bdr.node;
```

!!! Tip

> For more details about the `bdr.node` table, see the [PGD reference](/pgd/latest/reference/catalogs-visible#bdrnode).

Reviewer: You have broken the link to a specific entry in the reference

Author: I'm sorry, I don't really understand how this breaks the link by changing the link text? Please explain as there might be a mechanism I'm not familiar with.

Reviewer: The link is from bdr.node to the reference documentation for bdr.node

But moving it to PGD reference, the link is broken between the actual text and the content and that goes forward in time polluting search engines and their perception of the text and the link (and making it look like a spammy replication).

Author: ```suggestion
See  the [`bdr.node`](/pgd/latest/reference/catalogs-visible#bdrnode) table for details.
```
How's this?

!!!

This command lists just the node names. If you need to know the group they are a member of, use:

```
select node_name, node_group_name from bdr.node_summary;
```

!!! Tip

> For more details about about the `bdr.node_summary` table, see the [PGD reference](/pgd/latest/reference/catalogs-visible#bdrnode_summary).

Reviewer: Link broken again.

Author: Same.

!!!

## Using `bdr.run_on_nodes()`

> PGD can run specific commands on specific nodes using the `bdr.run_on_nodes()` function. This function takes two parameters: an array of node names and the command you want to run on those nodes. For example:

Reviewer: This is introducing people to the fact that there is this capability. It's not widely known.

> PGD can run specific commands on specific nodes using the `bdr.run_on_nodes()` function. This function takes two parameters: an array of node names and the command you want to run on those nodes. For example:

Author: ```suggestion
PGD has the ability to run specific commands on specific nodes using the `bdr.run_on_nodes()` function. This function takes two parameters: an array of node names and the command you want to run on those nodes. For example:
```
I find that wordy, but if you think this makes a difference, we can revert.

```

This command runs the `vacuum full foo` command on the node-one and node-three nodes.

> The node names are passed to the function in an array.

Reviewer: Again we are drawing attention to this.

Author: "Note that" is classic doc wordiness and doesn't really add any meaning. Do you want to somehow be more elaborate like, "as opposed to blah blah, in this case the node names are..."

The `bdr.run_on_nodes` function reports its results as JSONB. 
The results include the name of the node and the response (or error message) resulting from running the command.

> Other fields included might not be relevant. 

Reviewer: Not a useful contraction (the original was missing an and)

> Other fields included might not be relevant. 

Author: ```suggestion
Other fields included might be include and might not be relevant. 
```
Ah, I wasn't sure what the error was. I thought it was extra words.

> The results also appear as a single string that's hard to read. To make it more readable, apply some formatting.

Reviewer: We are about to go through the steps to apply formatting. You've changed it such that it suggests the reader should go and work out how to apply formatting.

Author: Okay, I'm not sure what you mean. Can you revise so it doesn't use first person?

Reviewer: By applying some formatting to this string, it can become more readable.

Author: ```suggestion
The results also appear as a single string that's hard to read. By applying some formatting to this string, it can become more readable.
```

$$ language 'plpgsql';
```

> This function takes a node name and a command and runs the command on that node, returning results like these:

Reviewer: These suggests what follows is just results when it's the command and output.

Author: I'm sorry, I'm not understanding what you mean. The "like so" was okay...It tonally didn't feel like doc, so I made this edit. I don't think we need to spend too much time on this, so if you think your original was somehow clearer, it was definitely an edit I didn't feel strongly about.

Reviewer: I was going for the "this is the interaction" - "these" suggests the following is all results.

Author: ```suggestion
This function takes a node name and a command and runs the command on that node, returning the results as shown in this interaction:
```
Is that what you mean? If not, let's just change it back to "like so".
